### ISSUE TITLE
Toggling Font Settings Doesn't Take Effect Until App Restart

### ISSUE THREAD

**Issue by andreymal** [mozilla-mobile/fenix#8735]
Firefox and Chrome are actually absolutely unusable on Android because the font-size behavior is inconsistent: 16px in some block can be bigger on smaller than 16px in another block.

I tried to toggle Accessibility options, but font-size is still inconsistent.

At today Opera for Android is the best browser because 16px is always 16px here. It draws desktop pages in a consistent and usable way.

I can enable this behavior using the "Desktop site" checkbox in the menu, but this is manual and uncomfortable. Opera does this automatically.

I want to have a option to enable this behavior **automatically**  in Firefox / Fenix, then I can migrate from Opera.

You can say the font is too small, but it **must** be small, because this is a **desktop** page. Zoom and [word wrapping](https://github.com/mozilla-mobile/fenix/issues/8736) should increase convenience.

Example:

```css
.div1 {
  font-size: 16px;
  float: left;
  width: 20%;
}
.div2 {
  font-size: 16px;
  float: right;
  width: 70%;
}
```

```html
<div class="div1">Sed ut perspiciatis unde omnis iste natus error...</div>
<div class="div2">Sed ut perspiciatis unde omnis iste natus error...</div>
```

Actual / expected behavior (Fenix / Opera) (click to view):
[![](https://i.imgur.com/HOuYtZfm.png)](https://i.imgur.com/HOuYtZf.png) [![](https://i.imgur.com/zxa6aI7m.png)](https://i.imgur.com/zxa6aI7.png)

**Comment by kbrosnan** [comment]
If you want consistent font sizes on mobile then you should tell the web engine that you are a mobile webpage. https://developer.mozilla.org/en-US/docs/Mozilla/Mobile/Viewport_meta_tag and https://developer.mozilla.org/en-US/docs/Web/Guide/Mobile

 If you don't control the webpage you can use the `Accessibility` settings to disable` Automatic font sizing` and set the `Font size` to `50%`.

**Comment by andreymal** [comment]
@kbrosnan font size 50% does not work because it breaks mobile websites, please reopen

**Comment by andreymal** [comment]
@kbrosnan font size 50% is actually not working even for desktop websites because the font size is still inconsistent (see Wikipedia as an example), please reopen ×2

**Comment by andreymal** [comment]
@kbrosnan there is a screenshot of Fenix with font size 50%. As you can see, the font size is still inconsistent.

Opera looks good and usable with zooming.

Note that the "Desktop site" checkbox is **disabled** in both browsers.

[![](https://i.imgur.com/bOWbPfbm.png)](https://i.imgur.com/bOWbPfb.png) [![](https://i.imgur.com/ucBowxBm.png)](https://i.imgur.com/ucBowxB.png)

**Comment by andreymal** [comment]
Oops, I got it... The reason why this did not work for me is that the 'Automatic Font Sizing' checkbox change is not applied until restart. Today I rebooted my phone and noticed that Fenix has the exact behavior that I want.

It looks like a bug, and it seems I should create a separate issue about it.

So this feature request is already implemented (with bugs) and this issue can actually be closed. Sorry for this flood, it was caused by the bug

![](https://i.imgur.com/xHeEDyO.gif)
